assets/src/apps/delivery/components/ActivityRenderer.tsx

```diff
@@ -420,7 +420,33 @@ const ActivityRenderer: React.FC<ActivityRendererProps> = ({
     }
     notifyStateMutation();
   }, [mutationTriggered]);
+  const debounceStateChanges = async (changes: any) => {
+    if (!ref.current) {
+      return;
+    }
+    const currentStateSnapshot: any = {};
+    if (changes.changed.length > 1 && Array.isArray(changes.changed)) {
```

Comment:
seems like if we need to check if it is an array that should happen before the length check, but maybe we don't need to check